### ISSUE TITLE
Add rolling window validation

### ISSUE THREAD

**Issue by safayavatsal** [codecollab-co/infra-cost#98]
**Epic**: #67 - Advanced ML-Based Cost Forecasting
**Week**: 6
**Estimate**: 5 hours

## Description
Implement rolling window validation for robust model accuracy assessment.

## Implementation
```typescript
class RollingWindowValidator {
  validate(
    data: CostDataPoint[],
    model: ForecastModel,
    windowSize: number,
    stepSize: number
  ): ValidationResult[] {
    const results: ValidationResult[] = [];
    
    for (let i = windowSize; i < data.length - stepSize; i += stepSize) {
      const trainData = data.slice(0, i);
      const testData = data.slice(i, i + stepSize);
      
      const forecast = model.train(trainData).predict(stepSize);
      const metrics = this.calculateMetrics(forecast, testData);
      
      results.push({
        windowStart: trainData[0].timestamp,
        windowEnd: trainData[trainData.length - 1].timestamp,
        testStart: testData[0].timestamp,
        testEnd: testData[testData.length - 1].timestamp,
        metrics
      });
    }
    
    return results;
  }
}
```

## Tasks
- [ ] Implement rolling window logic
- [ ] Support different window/step sizes
- [ ] Calculate metrics per window
- [ ] Aggregate results
- [ ] Visualize validation results
- [ ] Write unit tests
- [ ] Document methodology

## Acceptance Criteria
- Rolling window validation works
- Multiple windows tested
- Metrics aggregated correctly
- Tests validate logic
